Core/clim-core/frames/define-application-frame.lisp

```diff
@@ -213,6 +215,10 @@
          (:default-initargs
           :name              ',name
           :pretty-name       ,pretty-name
+          ,@(when icon
+              `(:icon (typecase ,icon
+                        ((or string pathname) (clim:make-pattern-from-bitmap-file ,icon))
+                        (t ,icon))))
```

Comment:
this should still allow specifying :icon nil (if someone wants to /not/ include default icons).

Reply:
ah, it does allow it already, nvm.

Reply:
Right. Also, I now think that coercing strings and pathnames should maybe be done in an `initialize-instance` method?

Reply:
makes sense to me.

Reply:
The macro now uses `icon-supplied-p`, all coercing is now done in the `initialize-instance` method.